### PR TITLE
Observations api

### PR DESCRIPTION
**Pull request by tinashechiraya**
Description:

I have fixed the gis/openlayers.html templates missing in django admin thereby preventing a user from creating a site

The observations now have a view to:
fetch all observations
fetch recent observations for the home page
fetch observation by id
create observation

clicking on an observation on the home page will now fetch the additional observation details which the user can see on the observation form on the map

https://github.com/kartoza/miniSASS/assets/70011086/ead2fc24-5060-4138-9256-5470dce54037

https://github.com/kartoza/miniSASS/assets/70011086/e31b7a70-b91c-4e97-9235-95d3cf60f582

still todo 
- [ ] populate the score form with observation data from api
- [ ] zoom to location on map when observation is clicked from home page 
- [ ] fix user profile creation 
I will do these in a seperate pr

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 3 min 37 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 4 sec

**Comment by dimasciput** [comment]
@tinashechiraya please add screenshot or a video.

@dimasciput I have attached them on the description

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 50 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 6 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 37 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 47 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 17 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 46 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 21 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 3 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 2 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 4 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 6 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 10 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 47 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 27 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 45 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 23 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 19 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 16 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 52 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 43 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 54 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 38 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 51 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 59 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 36 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 3 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 5 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 13 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 25 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 47 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 59 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 38 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 31 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 43 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 17 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 51 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 19 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 1 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 31 sec

File: django_project/monitor/observation_views.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        serializer = self.get_serializer(queryset, many=True)
+        recent_observations = []
+
+        for recent_observation in queryset:
+            try:
+                user_profile = UserProfile.objects.get(user=recent_observation.user)
+            except UserProfile.DoesNotExist:
+                user_profile = None
+
+            recent_observations.append({
```
Comment (dimasciput) [inline django_project/monitor/observation_views.py:32]: @tinashechiraya please use serializer

File: django_project/monitor/observation_views.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                user_profile = None
+
+            # Combine first name and last name
+            collectors_name = (
+                f"{user_profile.user.first_name} {user_profile.user.last_name}"
+                if user_profile and user_profile.user.first_name and user_profile.user.last_name
+                else user_profile.user.username if user_profile else ""
+            )
+            
+            observation_data = {
```
Comment (dimasciput) [inline django_project/monitor/observation_views.py]: @tinashechiraya please use serializer

**Review by dimasciput — CHANGES_REQUESTED** [review]
@tinashechiraya please add tests for the observation api in this PR

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 23 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 59 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 4 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 19 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 2 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 34 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 24 sec

File: django_project/monitor/urls.py [hunk 3]
```diff
@@ -13,13 +30,24 @@
     path('sites/<str:x>/<str:y>/<str:d>/', get_sites),
     path('closest_site/<str:x>/<str:y>/<str:d>/', get_closest_site),
     path('unique/<str:field>/', get_unique),
-    path('observation/<int:obs_id>/', zoom_observation),
-    path('observations/<int:site_id>/', get_observations),
+    path('observation-old/<int:obs_id>/', zoom_observation),
+    path('observations-old/<int:site_id>/', get_observations),
```
Comment (dimasciput) [inline django_project/monitor/urls.py]:
```suggestion
    path('legacy/observation/<int:obs_id>/', zoom_observation),
    path('legacy/observations/<int:site_id>/', get_observations),
```

**Review by dimasciput — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 22 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 44 sec

**Pull request merged**